src/rinex.c

```diff
@@ -1014,8 +1014,8 @@ static int readrnxobsb(FILE *fp, const char *opt, double ver, int *tsys,
             data[n].time=time;
             data[n].sat=(uint8_t)sats[i-1];
             
-            /* decode obs data */
-            if (decode_obsdata(fp,buff,ver,mask,index,data+n)&&n<MAXOBS) {
+            /* decode RINEX observation data */
+            if (decode_obsdata(fp,buff,ver,mask,index,data+n)) {
```

Comment:
any reason why we are dropping the second condition?

Reply:
It does not exist upstream